File: services/nuldoc/nuldoc-src/generators/slide_list.ts
import SlideListPage from "../pages/SlideListPage.ts";
import { Config } from "../config.ts";
import { Page } from "../page.ts";
import { SlidePage } from "./slide.ts";

export type SlideListPage = Page;

export async function generateSlideListPage(
  slides: SlidePage[],
  config: Config,
): Promise<SlideListPage> {
  const html = await SlideListPage(slides, config);

  return {
    root: html,
    renderer: "html",
    site: "slides",
    destFilePath: "/slides/index.html",
    href: "/slides/",
  };
}

File: services/nuldoc/lib/nuldoc/generators/slide_list.rb
module Nuldoc
  module Generators
    class SlideList
      def initialize(slides, config)
        @slides = slides
        @config = config
      end

      def generate
        html = Pages::SlideListPage.render(slides: @slides, config: @config)

        Page.new(
          root: html,
          renderer: :html,
          site: 'slides',
          dest_file_path: '/slides/index.html',
          href: '/slides/'
        )
      end
    end
  end
end
